[#93688938] Dynamic named GCE environments

mikkoc opened the pull request:
[Create independent environments automatically](https://www.pivotaltracker.com/n/projects/1275640/stories/93688938)

This PR implements the possibility to create isolated and independent GCE environments, based on a parameter passed to Terraform. e.g.:
`terraform apply -var 'env=prod'`

I paired with @jimconner on this.

Commit:
Add unique identifier for dynamic environments in GCE

Previously, we were not able to create different environments from the same template.
This was due to resource naming conflicts.
We are now making the resource names unique per environement. E.g: `tsuru-router-dev`, `tsuru-router-prod`
The environment name is passed to Terraform as a parameter when creating the infrastructure, e.g:
`terraform apply -var 'env=prod'`

Commit:
Use a more consistent naming convention

This commit does not change any logic, just the naming of the Terraform resources to be more consistent.

dcarley commented on gce/api-servers.tf:
```diff
@@ -1,7 +1,7 @@
 resource "google_compute_instance" "api" {
   count = 2
-  name = "tsuru-api-${count.index}"
+  name = "tsuru-api-${var.env}-${count.index}"
```
Nitpick, I think this should be the other way round. So that we know the environment is always the suffix.

dcarley commented on gce/routers.tf:
```diff
@@ -1,7 +1,7 @@
 resource "google_compute_instance" "router" {
   count = 2
-  name = "tsuru-router-${count.index}"
+  name = "tsuru-router-${var.env}-${count.index}"
```
Ditto comment above about order of vars.

dcarley commented on gce/ssl-proxies.tf:
```diff
@@ -1,7 +1,7 @@
+resource "google_compute_instance" "sslproxy" {
   count = 2
-  name = "tsuru-sslproxy-${count.index}"
+  name = "tsuru-sslproxy-${var.env}-${count.index}"
```
Ditto comment above about order of vars.

dcarley commented:
Other than that one thing, :+1:

mikkoc commented:
tbh, I'd be more inclined in removing the `${count.index}` all together since it does not add any value anyway.

jimconner commented:
I'd vote for using a prefix rather than a suffix as then all the VMs will be grouped together in the GCE web gui. (It's not such an issue on AWS where you can filter the list)

dcarley commented:
> tbh, I'd be more inclined in removing the ${count.index} all together since it does not add any value anyway.

I think we need it because each instance needs a unique name in GCE.

> I'd vote for using a prefix rather than a suffix as then all the VMs will be grouped together in the GCE web gui. (It's not such an issue on AWS where you can filter the list)

As in `prod-tsuru-api-0`?

jimconner commented:
Yes - exactly like that

mikkoc commented:
Ah you're right, my bad, got confused with AWS tags :)

dcarley commented:
> Yes - exactly like that

I'm inclined to agree. What do you think @mikkoc? And do we mind change AWS too?

dcarley commented:
Agreed to do this. Mikko and Jim are modifying.

jimconner committed:
Moved var.env from suffix to prefix

It will be easier to group VMs in the GCE api if we use a prefix instead of a suffix

dcarley commented:
Thanks :+1:
